### PR TITLE
Fix matching issues with playback tests

### LINKED ISSUE #976: [BUG] ChatClientDVRTest requests fail to match the request in the cassette
Running the DVR-based tests for Chat in playback mode fails because the requests don't match what is persisted in the cassette. This seems to be an issue with how the request URLs are either filtered or persisted.

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #976.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -52,7 +52,6 @@ target 'AzureCommunicationChat' do
     inherit! :search_paths
     pod 'OHHTTPStubs/Swift'
     pod 'Trouter', '0.0.1-beta.5'
-    pod 'MSAL', '1.1.15'
```
Comment (jairmyree) [inline Podfile:55]: What is this pod and there a reason why it is being removed?

File: sdk/communication/AzureCommunicationChat/Tests/ChatClientUnitTests.swift [hunk 2]
```diff
@@ -39,9 +38,9 @@ class ChatClientUnitTests: XCTestCase {
...
 
     override func setUpWithError() throws {
-        let settings = TestSettings.loadFromPlist()
-        let endpoint = settings?.endpoint ?? "https://endpoint"
-        let token = settings?.token ?? generateFakeToken()
+        let settings = TestSettings()
+        let endpoint = settings.endpoint
+        let token = settings.token
         let credential = try CommunicationTokenCredential(token: token)
         let options = AzureCommunicationChatClientOptions()
```
Comment (jairmyree) [inline sdk/communication/AzureCommunicationChat/Tests/ChatClientUnitTests.swift:45]: This appears in `setupWithError()` for just about every Chat test class. Would this be worth refactoring? If so, I'm not sure if that would be something you'd want to do or leave for chat.

File: sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift [hunk 3]
```diff
@@ -28,8 +28,22 @@ import AzureTest
...
 class TestSettings: TestSettingsProtocol {
-    var endpoint: String?
-    var user1: String?
-    var user2: String?
-    var token: String?
+    var endpoint = "https://endpoint/"
+    var user1 = "user1"
+    var user2 = "user2"
+    var token: String = {
+        let fakeValue = "{\"iss\":\"ACS\",\"iat\": 1608152725,\"exp\": 1739688725,\"aud\": \"\",\"sub\": \"\"}"
```
Comment (jairmyree) [inline sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift:35]: Just curious, what format is this?

File: sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift [hunk 4]
```diff
@@ -28,8 +28,22 @@ import AzureTest
...
+            .base64EncodedString()
+        let token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9." + fakeValue + ".EMS0ExXqRuobm34WKJE8mAfZ7KppU5kEHl0OFdyree8"
```
Comment (jairmyree) [inline sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift:37]: Same here, just curious as to what the token format is

**Review by jairmyree — APPROVED** [review]
I have some questions but they're just informational.

File: Podfile (hunk 1 above)
Comment (tjprescott) [reply Podfile:55]: It's not used by Chat so there's no reason for it to be listed as a dependency.

File: sdk/communication/AzureCommunicationChat/Tests/ChatClientUnitTests.swift (hunk 2 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationChat/Tests/ChatClientUnitTests.swift:45]: I'd leave it for the chat folks, but yeah, they could refactor it.

File: sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift (hunk 3 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift:35]: JSON string representation of their token object.

File: sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift (hunk 4 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationChat/Tests/TestSettings.swift:37]: Some weird representation of their token. For our intents and purposes, it's just a string.

**Review by jairmyree — APPROVED** [review]

**Pull request merged**